### PR TITLE
fix: fix keyboard navigation for folded listbox and folded listbox dropdown

### PR DESCRIPTION
**Pull request by quanho**
To fix https://github.com/qlik-oss/sn-list-objects/issues/240.

### CODE REVIEW

File: packages/sn-filter-pane/src/components/ListboxGrid/ListboxGrid.tsx [hunk 1]
```diff
@@ -129,7 +129,7 @@ function ListboxGrid({ stores }: { stores: IStores }) {
   }, [resources]);
 
   useEffect(() => {
-    const firstChild = gridRef?.current?.querySelector && gridRef?.current?.querySelector('.listbox-container,.folded-listbox-container') as HTMLDivElement;
+    const firstChild = gridRef?.current?.querySelector && gridRef?.current?.querySelector('.listbox-container,.listbox-popover-container') as HTMLDivElement;
```
Comment (johanlahti) [inline packages/sn-filter-pane/src/components/ListboxGrid/ListboxGrid.tsx:132]:
```suggestion
    const firstChild = gridRef?.current?.querySelector?.('.listbox-container,.listbox-popover-container') as HTMLDivElement;
```

File: packages/sn-filter-pane/src/components/ListboxPopoverContainer.tsx [hunk 2]
```diff
@@ -76,8 +97,125 @@ export const ListboxPopoverContainer = ({ resources, stores }: FoldedPopoverProp
...
+    switch (event.key) {
+      case KEYS.ESC:
+        onEscape(event);
+        handleClose();
+        break;
+      case KEYS.ENTER:
+        onFoldedListboxDropdownEnter(event);
+        break;
+      case KEYS.UP:
+      case KEYS.DOWN:
```
Comment (johanlahti) [inline packages/sn-filter-pane/src/components/ListboxPopoverContainer.tsx:207]: Question: Should we also support left and right?
```
case KEYS.LEFT:
case KEYS.RIGHT:
```

Comment (quanho) [reply packages/sn-filter-pane/src/components/ListboxPopoverContainer.tsx:197]: This is dropdown so LEFT and RIGHT is not needed. But maybe.

File: packages/sn-filter-pane/src/components/FoldedListbox/FoldedListbox.tsx [hunk 3]
```diff
@@ -20,12 +20,28 @@ export interface FoldedListboxProps {
...
+}
+
+const StyledDiv = styled('div', { shouldForwardProp: (p) => !['isInPopover'].includes(p as string) })<StyledDivProps>(
+  ({ isInPopover }) => ({
+    '&:focus': {
+      boxShadow: 'inset 0 0 0 2px #3F8AB3 !important',
+    },
+    '&:focus-visible': {
+      outline: 'none',
+    },
```
Comment (T-Wizard) [inline packages/sn-filter-pane/src/components/FoldedListbox/FoldedListbox.tsx:41]: I have ment to asl.
do someone know why we do it like this and not ?
```
    '&:focus-visible': {
      boxShadow: 'inset 0 0 0 2px #3F8AB3 !important',
      outline: 'none',
    },
```

File: packages/sn-filter-pane/src/components/ListboxPopoverContainer.tsx (hunk 2 above)
Comment (T-Wizard) [reply packages/sn-filter-pane/src/components/ListboxPopoverContainer.tsx:197]: maybe for screen reader user. they don't see it and don't know if it is a up/down or left/right navigation.
supporting both could make it easier for them?

File: packages/sn-filter-pane/src/components/FoldedListbox/FoldedListbox.tsx (hunk 3 above)
Comment (quanho) [reply packages/sn-filter-pane/src/components/FoldedListbox/FoldedListbox.tsx:41]: https://pawelgrzybek.com/the-difference-between-css-focus-and-focus-visible-pseudo-class/#:~:text=Adding%20the%20%3Afocus%20pseudo%2Dclass,research%20in%20one%20CSS%20property.

**Pull request merged**